back/scripts/datasets/sirene.py

```diff
@@ -45,13 +46,96 @@ def __init__(self, config: dict):
     @tracker(ulogger=LOGGER, log_start=True)
     def run(self) -> None:
         self._fetch_zip()
+        self._fetch_xls_files()
         self._format_to_parquet()
 
     def _fetch_zip(self):
         if self.zip_filename.exists():
             return
         urllib.request.urlretrieve(self._config["url"], self.zip_filename)
 
+    def _fetch_xls_files(self):
+        xls_links = self._config.get("xls_urls_naf", [])
+        for file_url in xls_links:
+            file_name = file_url.split("/")[-1]
+            file_path = self.data_folder / file_name
+            if not file_path.exists():
+                urllib.request.urlretrieve(file_url, file_path)
+
+        xls_url_cat_ju = self._config.get("xls_url_cat_ju")
+
+        if xls_url_cat_ju:
+            file_name = xls_url_cat_ju.split("/")[-1]
+            file_path = self.data_folder / file_name
+            if not file_path.exists():
+                urllib.request.urlretrieve(xls_url_cat_ju, file_path)
+
+    def join_naf_level(
+        self,
+        base_df,
+        level,
+        nomenclature_filter_col="nomenclature_naf",
+        nomenclature_value="NAFRev2",
+    ):
```

Comment:
nit: let's add type hints 
```suggestion
    def join_naf_level(
        self,
        base_df: pl.DataFrame,
        level: str,
        nomenclature_filter_col="nomenclature_naf": str,
        nomenclature_value="NAFRev2": str,
    ) -> pl.DataFrame:
```